core/src/main/java/com/linecorp/armeria/server/docs/MethodInfo.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2015 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.docs;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableSortedSet.toImmutableSortedSet;
+import static java.util.Comparator.comparing;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Streams;
+
+import com.linecorp.armeria.common.http.HttpHeaders;
+import com.linecorp.armeria.server.Service;
+
+/**
+ * Metadata about a function of a {@link Service}.
+ */
+public final class MethodInfo {
+
+    // FIXME(trustin): Return types and exception types should also have docstrings like params have them.
+
+    private final String name;
+    private final TypeSignature returnTypeSignature;
+    private final List<FieldInfo> parameters;
+    private final Set<TypeSignature> exceptionTypeSignatures;
+    private final List<HttpHeaders> exampleHttpHeaders;
+    private final List<String> exampleRequests;
+    private final String docString;
+
+    /**
+     * Creates a new instance.
+     */
+    public MethodInfo(String name,
+                      TypeSignature returnTypeSignature,
+                      Iterable<FieldInfo> parameters,
```

Comment:
Naming wise, that seems fine to me. But even if we split MethodInfo, we'd also probably need to split the rendering logic in the UI, and it seems like it might be over-complicated, so leaving as is seems fine with me too.